apps/site/pages/en/learn/asynchronous-work/discover-promises-in-nodejs.md

```diff
@@ -367,21 +367,31 @@ console.log('Synchronous task executed');
 
 ### `setImmediate()`
 
-`setImmediate()` is used to execute a callback after the current event loop cycle finishes and all I/O events have been processed. This means that `setImmediate()` callbacks run after any I/O callbacks, but before timers.
+`setImmediate()` schedules its callback for the event loop’s check phase—i.e. it will run immediately after all of the current loop’s I/O callbacks (poll phase) have completed.
```

Comment:
```suggestion
`setImmediate()` is used to execute a callback immediately after all of the current loop's I/O callbacks have completed.
```

This article is for beginners. For more advanced readers, there's a link at the bottom to another article with the specifics.